heroic-core/src/main/java/com/spotify/heroic/cluster/TracingClusterNode.java

```diff
@@ -79,67 +88,70 @@ public ClusterNode node() {
 
         @Override
         public AsyncFuture<FullQuery> query(FullQuery.Request request) {
-            return delegateGroup.query(request).directTransform(FullQuery.trace(queryIdentifier));
+            return delegateGroup
+                .query(request)
+                .directTransform(FullQuery.trace(queryIdentifier))
+                .catchFailed(wrapExceptionTransform());
         }
 
         @Override
         public AsyncFuture<FindTags> findTags(FindTags.Request request) {
-            return delegateGroup.findTags(request);
+            return delegateGroup.findTags(request).catchFailed(wrapExceptionTransform());
```

Comment:
This can be done once (in ClusterShard) if you give TracingClusterNode a good `toString()` implementation.